### ISSUE TITLE
CRA + TS + Emotion = how to get it working? 

### ISSUE THREAD

**Issue by MariuzM** [emotion-js/emotion#1868]
I'm using Create React App + TypeScript and Emotion. 

To get them working this is what i have to do: 

App.tsx
```js
// eslint-disable-next-line
import React from 'react'
/** @jsx jsx */ import { jsx } from '@emotion/core'
import { css } from '@emotion/core'

const myTest = css`
  background-color: red;
  width: 400px;
  height: 400px;
  font-size: 10px;
`

function App(): JSX.Element {
  return <div css={myTest}>Testing Emotion</div>
}

export default App
```

babelrc.js
```js
module.exports = {
  presets: ['@emotion/babel-preset-css-prop'],
  plugins: [
    [
      'emotion',
      {
        sourceMap: false,
        autoLabel: true,
        labelFormat: '-',
        cssPropOptimization: true,
      },
    ],
    // '@babel/plugin-proposal-class-properties',
    // [
    //   '@babel/plugin-transform-runtime',
    //   {
    //     regenerator: true,
    //   },
    // ],
  ],
}
```

package.json
```json
{
  "name": "my-app",
  "version": "0.1.0",
  "private": true,
  "homepage": "./",
  "dependencies": {
    "@emotion/babel-preset-css-prop": "^10.0.27",
    "@emotion/core": "^10.0.28",
    "@types/react": "^16.9.34",
    "@types/react-dom": "^16.9.7",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-scripts": "3.4.1",
    "typescript": "~3.8.3"
  },
  "devDependencies": {
    "@craco/craco": "^5.6.4",
    "@typescript-eslint/eslint-plugin": "^2.31.0",
    "@typescript-eslint/parser": "2.x",
    "babel-eslint": "10.x",
    "eslint": "7.x",
    "eslint-config-prettier": "^6.11.0",
    "eslint-config-react-app": "^5.2.1",
    "eslint-plugin-flowtype": "4.x",
    "eslint-plugin-import": "2.x",
    "eslint-plugin-jsx-a11y": "6.x",
    "eslint-plugin-prettier": "^3.1.3",
    "eslint-plugin-react": "7.x",
    "eslint-plugin-react-hooks": "4.x",
    "prettier": "^2.0.5"
  },
  "scripts": {
    "dev": "craco start --config config/webpack.dev.js",
    "build": "craco build --config config/webpack.prod.js",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}
```

So the issues / things i don’t like are:
- In `App.tsx` file i have to add `// eslint-disable-next-line` because after adding ```js /** @jsx jsx */ import { jsx } from '@emotion/core' ``` i get `React` is undefined.  

Is there a better way of writing this in CRA? that i don't need to use `// eslint-disable-next-line` and  ```js /** @jsx jsx */ import { jsx } from '@emotion/core' ```? Emotion is working fine on my Next.js projects without all this and on my manual setup for React (not using CRA)

**Comment by Andarist** [comment]
>  i have to add // eslint-disable-next-line because after adding js /** @jsx jsx */ import { jsx } from '@emotion/core' i get React is undefined.

This seems like an issue with ESLint configuration or a bug in the used plugins.

As to the requirement of using pragma - CRA doesn't allow you to configure Babel. I see that you are using craco so maybe it would be possible for you to leverage `'@emotion/babel-preset-css-prop'` to avoid this, but it's hard to tell without you sharing the **runnable** repro case.

**Comment by MariuzM** [comment]
Yep using babelrc.js already still same issue :( 
```js
module.exports = {
  presets: ['@emotion/babel-preset-css-prop'],
  plugins: [
    [
      'emotion',
      {
        sourceMap: false,
        autoLabel: true,
        labelFormat: '-',
        cssPropOptimization: true,
      },
    ],
  ],
}
```

**Comment by MariuzM** [comment]
Repo link to tes: https://github.com/MariuzM/cra-ts-emotion-test

**Comment by Andarist** [comment]
I don't see any ESLint errors/warnings using this repo.

As to Babel config - I'm experienced with craco so there might be a better way, but I've managed to use your babelrc by using such `webpack.dev.js`:
```js
module.exports = {
  babel: require(__dirname + '/../.babelrc.js'),

  devServer: {
    writeToDisk: false,
  },
}
```

**Comment by MariuzM** [comment]
This is working fine on my webpack.dev.js 
```js
  babel: {
    loaderOptions: {
      babelrc: true,
    },
  },
```

Also there won't be ESLint errors at-least not now, but there are errors when run the app

**Comment by Andarist** [comment]
Could you rephrase? I'm a little bit confused now - it seems that your repro doesn't suffer from any problems now, so I'm not sure how can I help any further.

**Comment by MariuzM** [comment]
Apologies. Currently the issues i have are:
- I can't use the `<>` i have to use it with `<React.Fragment>` or `<Fragment>`. 
- On each component i have to declare this `/** @jsx jsx */`, i would like utilise emotion without declaring it.

**Comment by Andarist** [comment]
I've made minor changes and both issues are resolved with this patch:
```diff
diff --git a/.babelrc.js b/.babelrc.js
index eca3565..6982acb 100644
--- a/.babelrc.js
+++ b/.babelrc.js
@@ -1,14 +1,3 @@
 module.exports = {
-  presets: ['@emotion/babel-preset-css-prop'],
-  plugins: [
-    [
-      'emotion',
-      {
-        sourceMap: false,
-        autoLabel: true,
-        labelFormat: '-',
-        cssPropOptimization: true,
-      },
-    ],
-  ],
+  "presets": ["@emotion/babel-preset-css-prop"]
 }
diff --git a/config/webpack.dev.js b/config/webpack.dev.js
index 84bc94a..b197d47 100644
--- a/config/webpack.dev.js
+++ b/config/webpack.dev.js
@@ -1,9 +1,5 @@
 module.exports = {
-  babel: {
-    loaderOptions: {
-      babelrc: true,
-    },
-  },
+  babel: require(__dirname + '/../.babelrc.js'),
 
   devServer: {
     writeToDisk: false,
diff --git a/src/App.tsx b/src/App.tsx
index b5c4856..961c780 100644
--- a/src/App.tsx
+++ b/src/App.tsx
@@ -1,6 +1,5 @@
-/** @jsx jsx */
-import { Fragment } from 'react'
-import { css, jsx } from '@emotion/core'
+import * as React from 'react'
+import { css } from '@emotion/core'
 
 const myTest = css`
   background-color: red;
@@ -11,13 +10,9 @@ const myTest = css`
 
 function App(): JSX.Element {
   return (
-    <Fragment>
+    <>
       <div css={myTest}>Testing Emotion</div>
-
-      {/* <div>
-        <>Fragment not working</>
-      </div> */}
-    </Fragment>
+    </>
   )
 }
```

**Comment by MariuzM** [comment]
I believe updated everything but getting this error when running `ReferenceError: jsx is not defined` i believe this is due to not including `/** @jsx jsx */`

**Comment by Andarist** [comment]
Please clear your cache and try to apply this patch I've given 1 to 1.

**Comment by MariuzM** [comment]
Done it, still same. 
Deleted: node_modules and rebuild using `yarn install` 
Chrome: Dev mode + Cashe disabled.

**Comment by MariuzM** [comment]
> Done it, still same.
> Deleted: node_modules and rebuild using `yarn install`
> Chrome: Dev mode + Cashe disabled.

Could you send me your repo with patch?

**Comment by Andarist** [comment]
Please apply changes and push out a new commit - I'm going to do a fresh clone later to see what happens.

**Comment by MariuzM** [comment]
Done that also, mentioned previously. Also to clarify I'm using TS so maybe i still need to use `/** @jsx jsx */` and there is no away around it?

**Comment by Andarist** [comment]
You have not applied my patch 1 to 1. You have commented out some things but not removed them - this is not the same. Once you remove [this line](https://github.com/MariuzM/cra-ts-emotion-test/blob/6466c0ba16dbaf3011a0364d5174699a33fe2b47/src/App.tsx#L1) it starts to work.

**Comment by MariuzM** [comment]
I see thank you, thought commenting this will not make it work, my bad.
